test.browser-bootstrap.js

```diff
@@ -0,0 +1,40 @@
+const { spawn } = require('child_process')
+const puppeteer = require('puppeteer')
+
+function isOldNode () {
+  const match = process.version.match(/^v(\d+)/)
+  const v = Number(match[1])
+  return v < 10
+}
+
+function run () {
+  // airtap requires node 10
+  if (isOldNode()) return
+
+  const args = process.argv.slice(2)
+
+  const airtap = spawn('airtap', args, { stdio: ['inherit', 'inherit', 'pipe'] })
+  airtap.on('close', (code) => process.exit(code))
+
+  let errBuffer = ''
+  let foundUrl = false
+  airtap.stderr.on('data', (data) => {
+    process.stderr.write(data)
+
+    if (foundUrl) return
+
+    errBuffer += data.toString()
+    const match = errBuffer.match(/https?:\/\/localhost:\d+\/airtap/)
+    if (!match) return
+
+    const url = match[0]
+    foundUrl = true
+    puppeteer.launch().then(function (browser) {
```

Comment:
why do it this very manual way instead of something supported by airtap out-of-the-box, like https://github.com/airtap/playwright?

Reply:
I was not familiar with Playwright, and from the messaging around it assumed it’s a commercial product. On second look that looks better, indeed.

Reply:
To be clear, thanks for noticing this, I’m happy to learn about Playwright.

Reply:
np!